lib/internal/dns/utils.js

```diff
@@ -141,10 +141,20 @@ function validateHints(hints) {
   }
 }
 
+function emitInvalidHostnameWarning(hostname) {
```

Comment:
Can you please make this function emit the warning at most once? Thank you.

Reply:
Just find that most deprecation warnings are emitted only once. Of course.